src/test/scala/mesosphere/marathon/api/validation/AppDefinitionValidatorTest.scala

```diff
@@ -452,6 +452,23 @@ class AppDefinitionValidatorTest extends MarathonSpec with Matchers with GivenWh
     AppDefinition.validAppDefinition(to3).isSuccess should be(true)
   }
 
+  test("Resident app may not define acceptedResourceRoles") {
+    Given("A resident app definition")
+    val f = new Fixture
+    val from = f.validResident
+    AllConf.SuppliedOptionNames = Set("mesos_authentication_principal", "mesos_role", "mesos_authentication_secret_file")
+
+    When("validating with acceptedResourceRoles")
+    val to1 = from.copy(acceptedResourceRoles = Some(Set("foo")))
+    Then("Should be invalid")
+    AppDefinition.validAppDefinition(to1).isSuccess shouldBe false
+
+    When("validating without acceptedResourceRoles")
+    val to2 = from.copy(acceptedResourceRoles = None)
```

Comment:
The thing is that I explicitly **do not** use the applications acceptedResourceRoles for matching for resources to reserve, because there's no way to consider anything other than unreserved resources (unless we want to add support for statically reserved resources, which is not recommended by Mesos). So my initial inclination was to not allow setting it at all and adding a meaningful validation error. (I didn't add the latter because there are changes in the scheduler feature branch that would most probably have conflicted then).

Does that make sense to you?